apps/barista-design-system/src/pages/single-page/components/toc/toc.html

```diff
@@ -4,42 +4,50 @@
     (click)="_expandTocMenu()"
     [ngClass]="{ 'ba-is-active': _expandToc }"
   >
-    <a
-      href="#"
+    <p
       class="ba-toc-header ba-expandable-trigger ba-expandable-trigger-right"
       data-initialized="true"
-      >On this page</a
     >
+      On this page
+    </p>
     <div class="ba-toc-content ba-expandable-content">
       <ul class="ba-toc-list ba-list">
         <li
-          *ngFor="let heading of _headings$ | async"
+          *ngFor="let headline of tocItems"
           class="ba-toc-item"
           [class.ba-toc-item-active]="
-            _activeItems && _activeItems[0] === heading
+            _activeItems &&
+            _activeItems[0] &&
+            _activeItems[0].id === headline.id
           "
         >
           <a
             #headline
-            href="{{ heading.href }}"
+            id="{{ headline.id }}"
+            href="#{{ headline.id }}"
             class="ba-toc-link"
             (click)="_handleTocClick($event)"
-            >{{ heading.title }}</a
           >
-          <ul *ngIf="heading.hasSubheadlines" class="ba-toc-ul">
+            {{ headline.headline }}
+          </a>
+          <ul *ngIf="headline.children" class="ba-toc-ul">
             <li
-              *ngFor="let subheading of heading.subheadlines"
+              *ngFor="let subheadline of headline.children"
               class="ba-toc-item"
               [class.ba-toc-item-active]="
-                _activeItems && _activeItems[1] === subheading
+                _activeItems &&
+                _activeItems[1] &&
+                _activeItems[1].id === headline.id
               "
             >
               <a
                 #headline
-                href="{{ subheading.href }}"
+                id="{{ subheadline.id }}"
+                href="#{{ subheadline.id }}"
                 class="ba-toc-link"
                 (click)="_handleTocClick($event)"
-                >{{ subheading.title }}</a
+              >
+                {{ subheadline.headline }}</a
               >
```

Comment:
First of all we are using the router and we should leverage the routing instead of using hrefs that trigger full page reloads. https://angular.io/api/router/RouterLink#description

```html
<a
   [routerlink]="['/']"
   [fragment]="subheadline.id"
   [attr.id]="subheadline.id"
>{{ subheadline.headline }}</a>
```

But as far as I can remember we have our own router link component that is working with serverside rendering.

We don't want to trigger a page reload when clicking on this.

> Small Tipp: never do this in Angular `id="{{ subheadline.id }}"` always use bindings!

https://angular.io/guide/template-syntax#attribute-binding

Reply:
We have run into a lot of problems when using the routerLink (either our own or the one from angular) here, as it will always trigger the canActivate in the guard. The guard will reset the input in the page-service, which will force angular to rerender the whole page. I'd rather not use the routerLink here.